Sage/carbon
# refactor: convert filter-object-properties to ts

### Proposed behaviour

Convert the `filter-object-properties` util to typescript



### Current behaviour

Util is written in Javascript.



### Checklist




#### QA


### Additional context



### Testing instructions

Only basic regression tests should be needed, particularly that the 2 utils that use `filter-object-properties` still function as intended. Examples of specific components to test these utils on are given below:

1) `filter-styled-system-margin-props`: eg Card, check that any styled-system margin props provided to the component are applied to the `data-component="card"` wrapper

2) `filter-styled-system-padding-props`: eg PopoverContainer, check that any styled-system padding props provided to the component are applied to the `data-element="popover-container-content"` wrapper.

(There is also `filter-out-position-props` but this does not appear to be used in any components at the moment!)



The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
